editor/editor_file_dialog.cpp

```diff
@@ -294,7 +294,10 @@ void EditorFileDialog::_post_popup() {
 			if (res && name == "res://") {
 				name = "/";
 			} else {
-				name = name.get_file() + "/";
+			  if(name[name.length()-1] == '/'){
+			    name.erase(name.length()-1, 1);
+			  }
+			  name = name.get_file() + "/";		  
```

Comment:
Instead of using `erase`, you can append `"/"` *only* when it's not present yet.

You should also amend the commit to fix the code style (tabs for indentation, spaces around operators, trailing whitespace). Best practice when contributing to projects is to try to match the code style of nearby code. But we also use `clang-format` to enforce style, see https://docs.godotengine.org/en/latest/community/contributing/code_style_guidelines.html

Finally, this can crash if `name` is empty. This may or may not happen depending on the code that fills `recentd`, but it's still good practice to guard against a potential out of bounds read. The manual way would be `name.length() > 1 && name[name.length() - 1] == "/"`, but `String` also has more high level features that can simplify this code significantly: `name.right(1) == "/"` or even better `name.ends_with("/")`).

Reply:
It appears name.get_file() returns an empty string when the path ends in a slash, resulting in the related issue of the recent directories list omitting entries which end in the slash. AFAIK removing the ending slash via erase is necessary to ensure name.get_file() returns the file name.

Thank you for the advice! How's this as a revision?
```suggestion
				if(name.ends_with("/")){
					name.erase(name.length() - 1, 1);
				}
				name = name.get_file() + "/";		  
```
In an empty string, name.ends_with("/") returns false, so this code shouldn't be able to crash given an empty name string.

Reply:
Instead of removing a suffix to readd it, you can simply not add the suffix when it's not needed :)

```cpp
name = name.get_file();
if (!name.ends_with("/")) {
	name += "/";
}
```

Reply:
Thanks for the revision! Unfortunately, though, building with that code, I found that the bug was still present. Thus that code does not fix issue #49453.

AFAIK name.get_file() only works in cases where name doesn't end in a slash. If name starts-off ending in '/' before get_file() is run, then the path is recorded as "/" in the recent directories list. Therefore checking and removing an end slash appears to be required to fix this bug/allow get_file() to work correctly. That is, I believe that's the case.